sklearn/utils/testing.py

```diff
@@ -149,6 +149,13 @@ def assert_greater_equal(a, b, msg=None):
     assert a >= b, message
 
 
+def assert_no_nan(a, msg=None):
```

Comment:
`assert_all_finite` raises `ValueError` not `AssertionError`. As such it is a validation, not a testing, helper...